auth/src/main/java/com/firebase/ui/auth/IdpResponse.java

```diff
@@ -275,26 +290,35 @@ public String toString() {
                 ", mSecret='" + mSecret + '\'' +
                 ", mIsNewUser='" + mIsNewUser + '\'' +
                 ", mException=" + mException +
+                ", mPendingCredential=" + mPendingCredential +
                 '}';
     }
 
     @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
     public static class Builder {
         private final User mUser;
+        private final AuthCredential mPendingCredential;
 
         private String mToken;
         private String mSecret;
         private boolean mIsNewUser;
 
         public Builder(@NonNull User user) {
             mUser = user;
+            mPendingCredential = null;
+        }
+
+        public Builder(@NonNull AuthCredential pendingCredential) {
+            mUser = null;
+            mPendingCredential = pendingCredential;
```

Comment:
I'm not a huge fan of adding another failure state. I feel like this would make more sense just being in the exception itself. @samtstern What do you think?

Reply:
Yeah I am fine to just wrap it in the exception instead.